lib/ansible/runner.rb

```diff
@@ -169,106 +165,48 @@ def response(base_dir, ansible_runner_method, result)
-      def shared_params(extra_vars:)
-        {:cmdline => "--extra-vars '#{JSON.dump(extra_vars)}'"}
+        [ansible_runner_method, base_dir, :json, runner_args]
       end
 
       # Asserts passed parameters are correct, if not throws an exception.
-      #
-      # @param env_vars [Hash] Hash with key/value pairs that will be passed as environment variables to the
-      #        ansible-runner run
-      # @param extra_vars [Hash] Hash with key/value pairs that will be passed as extra_vars to the ansible-runner run
-      # @param playbook_path [String] Path to the playbook we will want to run
-      # @param roles_path [String] Path to the directory with roles
-      def validate_params!(env_vars, extra_vars, playbook_path, roles_path)
-        assert_hash!(env_vars)
-        assert_hash!(extra_vars)
-        assert_path!(playbook_path)
-        assert_path!(roles_path)
-      end
+      def validate_params!(env_vars, extra_vars, ansible_runner_method, playbook_or_role_args)
+        errors = []
 
-      # Asserts that passed argument is a hash
-      #
-      # @param hash [Hash] Passed argument we test for being Hash
-      def assert_hash!(hash)
-        unless hash.kind_of?(Hash)
-          raise "Passed parameter must be of type Hash, got: #{hash}"
-        end
-      end
+        errors << "env_vars must be a Hash, got: #{hash.class}" unless env_vars.kind_of?(Hash)
+        errors << "extra_vars must be a Hash, got: #{hash.class}" unless extra_vars.kind_of?(Hash)
 
-      # Asserts that passed path exists
-      #
-      # @param path [Hash, NilClass] Passed path we test for existence
-      def assert_path!(path)
-        return unless path
+        unless %w(run start).include?(ansible_runner_method.to_s)
+          errors << "ansible_runner_method must be 'run' or 'start'"
+        end
 
-        unless File.exist?(path)
-          raise "File doesn't exist: #{path}"
+        unless playbook_or_role_args.keys == [:playbook] || playbook_or_role_args.keys.sort == [:role, :role_skip_facts, :roles_path]
+          errors << "Unexpected playbook/role args: #{args}"
         end
-      end
 
-      # Returns ansible-runner executable command
-      #
-      # @param base_dir [String] ansible-runner private_data_dir parameter
-      # @param ansible_runner_method [String] Method we will use to run the ansible-runner. It can be either "run",
-      #        which is sync call, or "start" which is async call
-      # @return [String] ansible-runner executable command
-      def ansible_command(base_dir, ansible_runner_method)
-        "ansible-runner #{ansible_runner_method} #{base_dir} --json -i result --hosts localhost"
+        playbook = playbook_or_role_args[:playbook]
+        errors << "playbook path doesn't exist: #{playbook}" if playbook && !File.exist?(playbook)
+        role = playbook_or_role_args[:role]
+        errors << "role path doesn't exist: #{role}" if role && !File.exist?(role)
```

Comment:
I think this one is a valid test failure, previously this checked for the existence of `roles_path` and `role` is just the name of the role not a path.

Reply:
Yup my mistake.  This is why I love specs :D